### PR TITLE
Treat warnings as errors

### PR DESCRIPTION
**Pull request by cugone**
Normalized Warning Level for the test projects to Warning Level 4 as well.

----

Edit:
Related:
- Issue #528

### CODE REVIEW

File: test-graphics/TestGraphics.cpp [hunk 1]
```diff
@@ -40,13 +40,13 @@ NAS2D::State* TestGraphics::update()
 
 	{
 		const auto jitter = [&]() -> decltype(auto) {
-			std::uniform_int_distribution<int> dist(0, 64);
+			std::uniform_int_distribution<uint8_t> dist(0, 64);
 			return dist(generator);
 		};
-		for (int i = 0; i < 2000; ++i)
+		for (auto i = 0u; i < 2000u; ++i)
```
Comment (DanRStevens) [inline test-graphics/TestGraphics.cpp]: Uhh, was this really necessary? The loop variable isn't even used anywhere.

Though to be fair, I do tend to prefer unsigned when possible.

**Review by DanRStevens — COMMENTED** [review]
Should probably have separated the code fixes from the project setting adjustments. The code fixes could have been applied first.

**Comment by cugone** [comment]
Clicking too fast, I didn't realize the code needed to be changed until after I pushed the change.

File: test-graphics/TestGraphics.cpp (hunk 1 above)
Comment (cugone) [reply test-graphics/TestGraphics.cpp]: The test is kinda bad in the first place. I don't even know what it's supposed to be testing.

**Review by DanRStevens — APPROVED** [review]

**Pull request merged**